aarch64 platform mimx8mm/p update and new mimx8mn platform

JiafeiPan opened the pull request:
This patchset includes the following updates:
1.  Add new arm64 platform imx8mn platform support.
2. Clean up imx8mm and imx8mp Cortex-A platform support, each platform use two board configuration, one for non-SMP and the other for SMP kernel, both of them can boot from uboot command or Jailhouse inmate cell.
3. Enabled physical UART4 for all board configurations.
4. Update pinctrl_imx driver to be aarch64 compatible.
5. Enabled pinctrl for all these three platforms and it works with UART.

JiafeiPan committed:
boards: arm64: imx8mp: rename board name to smp and non-smp

1. In order to figure out board name easily for a-core and m-core, add
a53 suffix in a-core board names.
2. Will support two board configures: one for non-smp and one for smp, both
of them can boot from uboot command line or load into Jailhouse inmate, so
remove Jaihouse suffix from board names.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mp_evk: clean up dts

1. unify sram addres (0xC0000000) and size (use 1M bytes).
2. remoeve PSCI node from non-smp board.
3. update CPU node and uart node.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mp_evk: cleanup defconfig

Enable serial console and driver, enabled clock control, classify
configuration items.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
drivers: pinctrl: fix imx pinctrl 64bit compatible

Make imx pinctrl driver to be aarch64 compatible.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
soc: arm64: imx8mm/p: update SoC part number and name

To be synced with drivers in hal_nxp.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
soc: arm64: imx8mm/p: enable pinctrl for soc and boards

Enable pinctrl on imx8mm and imx8mp platforms.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mm: rename boards name to smp and non-smp

1. In order to figure out board name easily for a-core and m-core, add
   a53 suffix in a-core board names.

2. Will support two board configures: one for non-smp and one for smp, both
   of them can boot from uboot command line or load into Jailhouse inmate,
   so remove Jaihouse suffix from board names.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mm: update dts for CPU and uart

1. Enable all cpu cores in soc dts and disable unused core in board dts.
2. Use default irq priority for uart interrupt.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mm: cleanup and update defconfig

Enabled serial console, serial driver, clock control.
Classify configuration items.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mm/p: unify soc name in configuration item

Used unified name "IMX8M*" to replace "MIMX8M*".

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: imx8mm/p: update document accordingly

Update document for imx8mm and imx8mp platform.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: arm64: add imx8mn board support

i.MX8M Nano LPDDR4 EVK board is based on NXP i.MX8M Nano
applications processor, composed of a quad Cortex®-A53 cluster
and a single Cortex®-M47 core.

Zephyr OS is ported to run on the Cortex®-A53 core.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

carlocaione reviewed (commented):
Nice work. Just a comment about commit messages: please add one even if it a short one.

gmarull commented on soc/arm64/nxp_imx/imx8m/soc.h:
```diff
@@ -0,0 +1,27 @@
+/*
```
soc.h is gone for arm64, please remove it.

JiafeiPan replied:
Hi, @gmarull , thanks for your comments, but there are still some drivers need soc.h, such as:
drivers/clock_control/clock_control_mcux_ccm.c
drivers/pinctrl/pinctrl_imx.c
drivers/serial/uart_mcux_iuart.c

So how to handle this issue? just create a dumy one? thanks.

JiafeiPan replied:
Hi, @gmarull , thanks for your comments, but there are still some drivers need soc.h, such as:
drivers/clock_control/clock_control_mcux_ccm.c
drivers/pinctrl/pinctrl_imx.c
drivers/serial/uart_mcux_iuart.c

So how to handle this issue? just create a dumy one? thanks.

gmarull replied:
then please fix drivers to include the right headers.

JiafeiPan replied:
okay, I removed soc.h from this PR, and created another PR to fix drivers:
https://github.com/zephyrproject-rtos/zephyr/pull/49734

Thanks.

JiafeiPan commented:
> Nice work. Just a comment about commit messages: please add one even if it a short one.

Thanks your comments, will update it.

JiafeiPan committed:
drivers: clock_control: remove soc.h from mcux_ccm

soc.h has been removed for ARM64 SoC platforms and it is also needed by
ARM32, so remove it from related drivers.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
drivers: pinctrl: remove soc.h from pinctrl_imx

soc.h has been removed for ARM64 SoC platforms and it is also needed by
ARM32, so remove it from related drivers.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
drivers: serial: remove soc.h from iuart

soc.h has been removed for ARM64 SoC platforms and it is also needed by
ARM32, so remove it from related drivers.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan commented:
> Nice work. Just a comment about commit messages: please add one even if it a short one.

Done, thanks.

JiafeiPan committed:
west.yml: update hal_nxp module

This update for hal_nxp includes some update to enable ARM64 support.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

povergoing reviewed (dismissed):
LGTM

JiafeiPan commented:
Updated board document, please review again, thanks.

JiafeiPan committed:
boards: arm64: imx8mm/p: unify soc name

Replace imx8mm/p with mimx8m/p to unify name with the
other software components.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
CODEOWNERS: imx8m/p/n: update codeowner

Add me to be codeowner of platform imx8mm/p/n ARM64
platforms.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan commented:
Dear maintainers, update this PR with the following updates, thanks for your review:
Unify SoC name from imx8m/n/p to mimx8m/n/p with the name in hal_nxp.
Add me to be codeowner of platform imx8mm/p/n ARM64 platforms.

JiafeiPan commented:
As hal_nxp patches have been merged, so update west.yml to use actual commit ID.

JiafeiPan commented:
Dear maintainers, rebased, no dependency now, please review and merge it, thanks.
